SynCrtSock.pas

```diff
@@ -4881,18 +4885,33 @@ procedure TCrtSocket.SetInt32OptionByIndex(OptName, OptVal: integer);
 
 procedure TCrtSocket.OpenBind(const aServer, aPort: SockString;
   doBind: boolean; aSock: integer; aLayer: TCrtSocketLayer; aTLS: boolean);
-const BINDTXT: array[boolean] of string[4] = ('open','bind');
-      BINDMSG: array[boolean] of string = ('Is a server running on this address:port?',
-        'Another process may be currently listening to this port!');
+const
+  RETRYSLEEPTIME = 10;
+  RETRYCOUNT     = 5;
+  BINDTXT: array[boolean] of string[4] = ('open','bind');
+  BINDMSG: array[boolean] of string = ('Is a server running on this address:port?',
+                                       'Another process may be currently listening to this port!');
+var
+  aRetryCount:integer;
 begin
   fSocketLayer := aLayer;
   fWasBind := doBind;
+  aRetryCount:=0;
   if aSock<=0 then begin
     if (aPort='') and (aLayer<>cslUNIX) then
       fPort := DEFAULT_PORT[aTLS] else // default port is 80/443 (HTTP/S)
       fPort := aPort;
     fServer := aServer;
-    fSock := CallServer(aServer,Port,doBind,aLayer,Timeout); // OPEN or BIND
+    while (aRetryCount<RETRYCOUNT) do
+    begin
+      fSock := CallServer(aServer,Port,doBind,aLayer,Timeout); // OPEN or BIND
+      if fSock>0 then break;
+      if (NOT WSAIsFatalError) then
+      begin
+        sleep(RETRYSLEEPTIME);

```

Comment:
please make a comment to specify why waiting is required here (perhaps it should be the caller, or at least been an option, not always retry)